components/upload/interface.ts

```diff
@@ -58,7 +58,10 @@ export interface ZipButtonOptions {
   action?: string | ((file: NzUploadFile) => string | Observable<string>);
   directory?: boolean;
   openFileDialogOnClick?: boolean;
-  beforeUpload?(file: NzUploadFile, fileList: NzUploadFile[]): boolean | Observable<NzSafeAny>;
+  beforeUpload?(
+    file: NzUploadFile,
+    fileList: NzUploadFile[]
+  ): boolean | Observable<NzUploadFile | boolean> | Promise<NzUploadFile | boolean>;
```

Comment:
```suggestion
  ): boolean | Observable<NzUploadFile | File | Blob | boolean> | Promise<NzUploadFile | File | Blob | boolean>;
```

With this change, I think beforeUpload seems to be a good replacement for nzTransformFile. What do you think?

If so, we can deprecate nzTransformFile (since transformFile property does not exist in antd)